lib/DistributionProvider.pm

```diff
@@ -0,0 +1,16 @@
+package DistributionProvider;
+use strict;
+use warnings FATAL => 'all';
+use version_utils;
+
+use Distribution::Tumbleweed;
+use Distribution::Sle15Sp1;
+use Distribution::Sle12Sp4;
+
+sub provide {
+    return Distribution::Tumbleweed->new() if version_utils::is_tumbleweed();
+    return Distribution::Sle15Sp1->new() if version_utils::is_sle('=15-SP1');
+    return Distribution::Sle12Sp4->new() if version_utils::is_sle('=12-SP4');
```

Comment:
I guess we can still define ranges, as not we'll get undef for Leap and SP5, which ATM is same as SP4.

Reply:
For the prototype it was enough to introduce TW and both Sle versions. 
For working solution, just one line with Leap distri will do the trick.return Distribution::Leap15_1->new() if version_utils::is_leap('=15.1');

Reply:
I would simply put `return Distribution::Tumbleweed->new();` in the end. Also, soften conditions to `15-SP1+` and `12-SP4+`

Reply:
As for naming convention, I would use something like Distribution::SLE::12::SP4 Distribution::SLE::15::SP1 or without extra layer Distribution::SLE::12SP4, but I guess there will be common part between all service packs, so for SP0 we'll use  Distribution::SLE::12. WDYT? We also could reuse VERSION variable to find appropriate perl module.